test_scripts/Smoke/API/023_UnsubscribeButton_PositiveCase.lua

```diff
@@ -0,0 +1,113 @@
+---------------------------------------------------------------------------------------------------
+-- User story: Smoke
+-- Use case: UnsubscribeButton
+-- Item: Happy path
+--
+-- Requirement summary:
+-- [UnsubscribeButton] SUCCESS: getting SUCCESS:UnsubscribeButton()
+--
+-- Description:
+-- Mobile application sends valid UnsubscribeButton request and gets UnsubscribeButton "SUCCESS" response from SDL
+
+-- Pre-conditions:
+-- a. HMI and SDL are started
+-- b. appID is registered and activated on SDL
+-- c. appID is currently in Background, Full or Limited HMI level
+
+-- Steps:
+-- appID requests UnsubscribeButton with valid parameters
+
+-- Expected:
+-- SDL validates parameters of the request
+-- SDL checks if Buttons interface is available on HMI
+-- SDL checks if UnsubscribeButton is allowed by Policies
+-- SDL checks if all parameters are allowed by Policies
+-- SDL sends the Buttons notificaton to HMI
+-- SDL responds with (resultCode: SUCCESS, success:true) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSmoke = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Local Variables ]]
+local isMediaApplication = commonSmoke.GetAppMediaStatus()
+
+local buttonName = {
+  "OK",
+  "SEEKLEFT",
+  "SEEKRIGHT",
+  "TUNEUP",
+  "TUNEDOWN",
+  "PRESET_0",
+  "PRESET_1",
+  "PRESET_2",
+  "PRESET_3",
+  "PRESET_4",
+  "PRESET_5",
+  "PRESET_6",
+  "PRESET_7",
+  "PRESET_8"
+}
+
+--[[ Local Functions ]]
+local function checkResults(pCid, pSuccess, pResultCode, self)
+  self.mobileSession1:ExpectResponse(pCid, { success = pSuccess, resultCode = pResultCode })
+  if pResultCode == "SUCCESS" then
+    self.mobileSession1:ExpectNotification("OnHashChange")
+    :Times(1)
+  else
+    self.mobileSession1:ExpectNotification("OnHashChange")
+    :Times(0)
+  end
+end
+
+local function subscribeButtons(pButName, self)
+  local cid = self.mobileSession1:SendRPC("SubscribeButton", { buttonName = pButName })
+  local appIDvalue = commonSmoke.getHMIAppId()
+  EXPECT_HMINOTIFICATION("Buttons.OnButtonSubscription", { appID = appIDvalue, name = pButName, isSubscribed = true })
+  :Timeout(1000)
```

Comment:
@GetmanetsIrina Is it required to define explicit timeout here?

Reply:
@dboltovskyi , Removed in 2af89c88d07e5107b027dd7581cd618ef9f0ec90.